quantum/rgblight.c

```diff
@@ -185,6 +185,18 @@ void rgblight_update_dword(uint32_t dword) {
   }
 }
 
+void rgblight_update_dword_noeeprom(uint32_t dword) {
```

Comment:
Yeah, I asked in the firmware channel on Discord about this...

me:

> i was originally going to just add an write_to_eeprom parameter and modify api.c, but that doesn't seem nice to anyone else using it, so i was thinking of doing the whole _eeprom_helper() and _noeeprom() thing

drashna: 

> normally, I'd agree with you, but it appears that this function is literally only called in the api.c file, and nowhere else, except now in the split util code
> it may be simpler to just add a noeeprom version of the function in the rgblight.c, a reference in the rgblight.h file and call that instead

Reply:
I'm not suggesting adding a (API breaking) parameter or a helper function, more a removal of the duplicated code (for easier forward maintainability, less likely to have a change accidentally introduced in only one of the two places, etc) like:

````
void rgblight_update_dword(uint32_t dword) {
  eeconfig_update_rgblight(dword);
  rgblight_update_dword_noeeprom(dword);
}

void rgblight_update_dword_noeeprom(uint32_t dword) {
  rgblight_config.raw = dword;
  if (rgblight_config.enable)
    rgblight_mode_noeeprom(rgblight_config.mode);
  else {
    #ifdef RGBLIGHT_ANIMATIONS
      rgblight_timer_disable();
    #endif
      rgblight_set();
  }
}
````

Or is there some reason why this would be frowned upon?